crates/core/tedge/src/cli/certificate/cli.rs

```diff
@@ -70,8 +74,21 @@ impl BuildCommand for TEdgeCertCli {
                 };
                 cmd.into_boxed()
             }
+            TEdgeCertCli::Renew => {
+                if std::fs::metadata(config.device.cert_path.clone()).is_ok() {
```

Comment:
I would move this check inside `RenewCertCmd::renew_test_certificate()`:

- where the error type can be more appropriate (a specific `CertError` and not a generic `ConfigError`)
- and where the check can be deeper (checking notably that the certificate is actually self-signed)

If you go in that direction, I would also remove the `id` field from `RenewCertCmd` as this id can be extracted from the certificate (this is the subject CN) while checking the certificate is self signed. See `ShowCertCmd::show_certificate()`.

Reply:
refactored the code